### ISSUE TITLE
[Feedback] Create and Edit Tiers 

### ISSUE THREAD

**Issue by BenJam** [opencollective/opencollective#6092]
We've redesigned the process to create and edit tiers. Tiers can be used to provide perks and rewards to supporters, set membership fees, build funding drives, or simply categorise your contributions. The new experience shows a preview of the tier as it will appear on your Collective's profile:

![Screenshot 2022-11-01 at 10 07 45](https://user-images.githubusercontent.com/158833/199209956-f01e6019-f4f7-42a5-91fd-f970d9fd6ad6.png)

📘 [Read more about tiers and goas](https://docs.opencollective.com/help/collectives/collective-settings/tiers-goals)
📧 [Contact support if something isn't working](mailto:support@opencollective.com)
🙃 Leave feedback here!

To try the feature while it's in beta, add `-revamp` to the edit tiers URL, e.g.: https://opencollective.com/engineering/admin/tiers -> https://opencollective.com/engineering/admin/tiers-revamp.

**Comment by BenJam** [comment]
Initial feedback (from staging site)

- button does not appear in the preview — this is key in my opinion.
- [standalone page for new tier did not route correctly ](https://staging.opencollective.com/octoboxes/contribute/t-shirt-tier-k3z8arxv-eymko60y-w9mqgl5n-bj9w704d)

**Comment by Betree** [comment]
Adding my Slack feedback here:

- The goals here are not set properly, we need to fetch the stats. See https://staging.opencollective.com/captainfact_io for an example of how it should look like.
![image](https://user-images.githubusercontent.com/1556356/199217006-1f2b4d97-e12c-49ed-be28-4c12b7f0159b.png)

- :icecream: When creating/editing a tier, I'd like to see a small toast (addToast) in the bottom right like "Tier created|edited successfully".

- When alone on a line, the "Create tier" card gets smaller. We should ideally keep the same size
![image](https://user-images.githubusercontent.com/1556356/199217554-095a1641-bdd7-46cd-9fa3-b298dcaf1cdb.png)

- After saving a tier, its order in the list changes. It's a bit confusing cause you're looking at the place where you clicked on "Edit" and there's something else now
![Peek 2022-09-30 09-06](https://user-images.githubusercontent.com/1556356/199217621-63fc8b67-3018-4c0c-b37c-946aa3c2daa4.gif)

- The delete action should prompt a ConfirmationModal, it's easy to missclick the button. This modal will also be a good place to implement the "Cancel recurring contirbutions?" checkbox

- Some labels are using a different case
![image](https://user-images.githubusercontent.com/1556356/199217768-30b81e77-ba87-4bba-a528-1c7122e8c1b8.png)

- API errors don't seem to be surfaced with a toast, so it's silently failing. Here I'm getting a `"Variable \"$tier\" got invalid value null at \"tier.amount\"; Expected non-nullable type \"AmountInput!\" not to be null."`  after changing the type from fixed to flexible and playing a bit with the amount presets.
![Peek 2022-09-30 09-17](https://user-images.githubusercontent.com/1556356/199217836-4baef5d7-896d-46c1-bf63-b7c60b601689.gif)

- On Chrome I get this horizontal scrollbar. We should replace overflow: scroll; with `overflow-y: scroll;`
![image](https://user-images.githubusercontent.com/1556356/199217886-9dedd033-57ef-4997-b410-ac3f59366951.png)

- We need to make sure we keep two separate sections for events: Tiers and Tickets

**Comment by shannondwray** [comment]
Docs issue - https://github.com/opencollective/opencollective/issues/6139
Release issue - https://github.com/opencollective/opencollective/issues/6140

**Comment by Betree** [comment]
On responsiveness, there's currently a horizontal scrollbar when the screen width is 1080px. 

![image](https://user-images.githubusercontent.com/1556356/204236795-db0582bc-331a-4abf-ab7f-d7730184e171.png)

**Comment by Betree** [comment]
The new tiers page is also impacted by https://github.com/opencollective/opencollective/issues/2954

**Comment by shannondwray** [comment]
I don't have access to this feature in order to document it

**Comment by Betree** [comment]
@shannondwray I've just added instructions on how to test the feature while it's in beta in the issue body:
Add `-revamp` to the edit tiers URL, e.g.: https://opencollective.com/engineering/admin/tiers -> https://opencollective.com/engineering/admin/tiers-revamp.

**Comment by shannondwray** [comment]
Do we have a date for when we think it will be made public? :)

**Comment by Betree** [comment]
@hdiniz recently pushed an update that fixed most of the feedback that was shared here. One last round of quality assertion and we should be good to go. @hdiniz do you feel like we can propose a release date for this one?

**Comment by hdiniz** [comment]
Yes, I think we have feature parity with the previous version of the page, we can enable that and gather more feedback.

**Comment by Betree** [comment]
@hdiniz All good to me for a public release, I just had another testing session and it worked great overall. There's still some feedback from https://github.com/opencollective/opencollective/issues/6092#issuecomment-1298351292 that we should ideally address when we have the time (e.g. the goal progress bar stuck to 0).

**Comment by Betree** [comment]
We're missing the tax info box in the modal, this one is quite important

![image](https://user-images.githubusercontent.com/1556356/210540115-3beb84ad-9c32-4bf1-aeca-a2a45f40468e.png)

**Comment by Betree** [comment]
A very common flow when editing the tiers is to go back to your collective page to see what they look like. Since we're not refetching the collective page query, **a full page refresh is currently required before seeing the changes**.

Possible solutions:
- Add the collective page query to `refetchQueries` (heavy, we should at least remove `awaitRefetchQueries` in this case)
- Re-fetch only what's needed with a dedicated query like

```gql
collective(slug: ...) {
  id
  settings
  tiers {
    id
    ...ContributeCardTierFields
  }
}
```

**Comment by Betree** [comment]
Released & iterated on, we'll track problems in individual issues
